test/envtest/gpudeviceplugin_controller_test.go

```diff
@@ -58,16 +62,53 @@ var _ = Describe("GpuDevicePlugin Controller", func() {
 				return len(fetched.Status.ControlledDaemonSet.UID) > 0
 			}, timeout, interval).Should(BeTrue())
 
-			By("updating image name successfully")
+			By("DaemonSet created successfully")
+			ds := &apps.DaemonSet{}
+			_ = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: ns, Name: "intel-" + "gpu" + "-plugin"}, ds)
+			Expect(ds.Spec.Template.Spec.NodeSelector).To(Equal(fetched.Spec.NodeSelector))
+
+			for _, dsInitContainer := range ds.Spec.Template.Spec.InitContainers {
+				Expect(dsInitContainer.Image).To(Equal(spec.InitImage))
+			}
+
+			By("updating GpuDevicePlugin successfully")
 			updatedImage := "updated-testimage"
+			updatedInitImage := "updated-testinitimage"
```

Comment:
would it make sense to drop InitImage and thet that it's gone:
```
Expect(ds.Spec.Template.Spec.InitContainers).To(HaveLen(0))
```

Reply:
So, we test both when it was changed and when it's gone, right?

Reply:
@mythi So, do we test both when it's changed and when it's gone? To me, it makes sense that we test it.

Then, we need to make other rounds for updating dp and ds. Do I do so?

Reply:
check does is increase the coverage and if it does then it makes sense

Reply:
Ok, then would it also make sense to test nodeselector in the same way to increase the coverage?

Reply:
@mythi Ok, then would it also make sense to test nodeselector in the same way to increase the coverage?

Reply:
+1